modules/gdscript/gdscript_parser.cpp

```diff
@@ -4224,17 +4308,80 @@ bool GDScriptParser::export_group_annotations(const AnnotationNode *p_annotation
 
 bool GDScriptParser::warning_annotations(const AnnotationNode *p_annotation, Node *p_target, ClassNode *p_class) {
 #ifdef DEBUG_ENABLED
+	if (is_ignoring_warnings) {
+		return true; // We already ignore all warnings, let's optimize it.
+	}
+
 	bool has_error = false;
 	for (const Variant &warning_name : p_annotation->resolved_arguments) {
-		GDScriptWarning::Code warning = GDScriptWarning::get_code_from_name(String(warning_name).to_upper());
-		if (warning == GDScriptWarning::WARNING_MAX) {
+		GDScriptWarning::Code warning_code = GDScriptWarning::get_code_from_name(String(warning_name).to_upper());
+		if (warning_code == GDScriptWarning::WARNING_MAX) {
 			push_error(vformat(R"(Invalid warning name: "%s".)", warning_name), p_annotation);
 			has_error = true;
 		} else {
-			p_target->ignored_warnings.push_back(warning);
+			int start_line = p_annotation->start_line;
+			int end_line = p_target->end_line;
+
+			switch (p_target->type) {
+#define SIMPLE_CASE(m_type, m_class, m_property)          \
+	case Node::m_type: {                                  \
+		m_class *node = static_cast<m_class *>(p_target); \
+		if (node->m_property != nullptr) {                \
+			end_line = node->m_property->end_line;        \
+		} else {                                          \
+			end_line = node->start_line;                  \
+		}                                                 \
+	} break;
+
+				// Can contain properties (set/get).
+				SIMPLE_CASE(VARIABLE, VariableNode, initializer)
+
+				// Contain bodies.
+				SIMPLE_CASE(FOR, ForNode, list)
+				SIMPLE_CASE(IF, IfNode, condition)
+				SIMPLE_CASE(MATCH, MatchNode, test)
+				SIMPLE_CASE(WHILE, WhileNode, condition)
+#undef SIMPLE_CASE
```

Comment:
I'm not sure about that definition, readability-wise. Especially since `m_type` is only used in the first line.
I would do something more akin this, but I may be wrong:

```suggestion
#define SIMPLE_CASE(m_class, m_property)          \
	m_class *node = static_cast<m_class *>(p_target); \
	if (node->m_property != nullptr) {                \
		end_line = node->m_property->end_line;        \
	} else {                                          \
		end_line = node->start_line;                  \
	}

				// Can contain properties (set/get).
				case Node::VARIABLE: { SIMPLE_CASE(VariableNode, initializer) } break;

				// Contain bodies.
				case Node::FOR: { SIMPLE_CASE(ForNode, list) } break;
				case Node::IF: { SIMPLE_CASE(IfNode, condition) } break;
				case Node::MATCH { SIMPLE_CASE(MatchNode, test) } break;
				case Node::WHILE { SIMPLE_CASE(WhileNode, condition) } break;
#undef SIMPLE_CASE
```

Reply:
I would leave it as it is, maybe only remove `Node::` from the macro for readability. While there is no need to use the check inside a complex case, in my opinion it is better to leave `case` inside the macro. If this changes, then we can refactor it easily.